sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -345,3 +346,100 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+def test_imputation_knn():
```

Comment:
I tested it.     
imputer = Imputer(missing_values=999, strategy="knn",
                      axis=0, n_neighbors=1, copy=False)
X5 = X5.astype(float)
X_impute = imputer.fit(X5).transform(X5)
assert_array_equal(X_true_1, X5)